src/internal/m365/collection/site/restore_test.go

```diff
@@ -73,64 +74,90 @@ func (suite *SharePointRestoreSuite) TestListCollection_Restore() {
 	defer flush()
 
-	service := createTestService(t, suite.creds)
-	listing := spMock.ListDefault("Mock List")
-	testName := "MockListing"
-	listing.SetDisplayName(&testName)
-	byteArray, err := service.Serialize(listing)
+	testName, lrh, destName, mockData := suite.setupDependencies("genericList")
+
+	deets, err := restoreListItem(ctx, lrh, mockData, suite.siteID, destName)
 	require.NoError(t, err, clues.ToCore(err))
+	assert.Equal(t, fmt.Sprintf("%s_%s", destName, testName), deets.SharePoint.ItemName)
 
-	destName := testdata.DefaultRestoreConfig("").Location
+	// Clean-Up
+	suite.deleteList(ctx, lrh, t, deets)
+}
 
-	listData, err := data.NewPrefetchedItemWithInfo(
-		io.NopCloser(bytes.NewReader(byteArray)),
-		testName,
-		details.ItemInfo{SharePoint: ListToSPInfo(listing, int64(len(byteArray)))})
-	require.NoError(t, err, clues.ToCore(err))
+func (suite *SharePointRestoreSuite) TestListCollection_Restore_invalidTemplate() {
+	t := suite.T()
 
-	r, err := readers.NewVersionedRestoreReader(listData.ToReader())
-	require.NoError(t, err)
+	ctx, flush := tester.NewContext(t)
+	defer flush()
 
-	mockData := &dataMock.Item{
-		ItemID: testName,
-		Reader: r,
-	}
+	_, lrh, destName, mockData := suite.setupDependencies(api.WebTemplateExtensionsListTemplateName)
 
-	lrh := NewListsRestoreHandler(suite.siteID, suite.ac.Lists())
-	deets, err := restoreListItem(ctx, lrh, mockData, suite.siteID, destName)
-	require.NoError(t, err, clues.ToCore(err))
-	assert.Equal(t, fmt.Sprintf("%s_%s", destName, testName), deets.SharePoint.ItemName)
+	_, err := restoreListItem(ctx, lrh, mockData, suite.siteID, destName)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), api.ErrInvalidTemplateError.Error())
+}
 
-	// Clean-Up
+func (suite *SharePointRestoreSuite) deleteList(
+	ctx context.Context, lrh listsRestoreHandler, t *testing.T, deets details.ItemInfo,
+) {
 	var (
-		builder  = service.Client().Sites().BySiteId(suite.siteID).Lists()
 		isFound  bool
 		deleteID string
 	)
 
-	for {
-		resp, err := builder.Get(ctx, nil)
-		assert.NoError(t, err, "getting site lists", clues.ToCore(err))
+	lists, err := lrh.ac.Client.Lists().GetLists(ctx, suite.siteID, api.CallConfig{})
+	assert.NoError(t, err, "getting site lists", clues.ToCore(err))
 
-		for _, temp := range resp.GetValue() {
-			if ptr.Val(temp.GetDisplayName()) == deets.SharePoint.ItemName {
-				isFound = true
-				deleteID = ptr.Val(temp.GetId())
+	for _, l := range lists {
+		if ptr.Val(l.GetDisplayName()) == deets.SharePoint.ItemName {
+			isFound = true
+			deleteID = ptr.Val(l.GetId())
 
-				break
-			}
-		}
-		// Get Next Link
-		link, ok := ptr.ValOK(resp.GetOdataNextLink())
-		if !ok {
 			break
 		}
-
-		builder = sites.NewItemListsRequestBuilder(link, service.Adapter())
 	}
 
 	if isFound {
 		err := lrh.DeleteList(ctx, deleteID)
 		assert.NoError(t, err, clues.ToCore(err))
 	}
 }
+
+func (suite *SharePointRestoreSuite) setupDependencies(listTemplate string) (
```

Comment:
Prefer standalone funcs instead of methods for test helpers, please.  That promotes better re-use than binding functions to specific suites. 
```suggestion
func setupDependencies(
  suite tester.Suite,
  ac api.Client,
  siteID string,
  creds account.M365Config,
  listTemplate string,
  ) (
```